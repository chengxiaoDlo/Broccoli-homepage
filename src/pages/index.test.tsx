import { render, screen } from '@testing-library/react';
import Home from './index';
import React from 'react';

describe("homepage test", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      }))
    });
  });

  test('render by snapshot', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot();
  })
})