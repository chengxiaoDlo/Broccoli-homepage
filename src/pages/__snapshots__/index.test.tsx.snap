// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage test render by snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <header>
      BROCCOLI & CO.
    </header>
    <main>
      <div>
        <h2>
          A better way
        </h2>
        <h2>
          to enjoy every day.
        </h2>
        <div
          class="desc"
        >
          Be the first to know when we launch
        </div>
        <button
          class="ant-btn ant-btn-primary ant-btn-lg"
          style="display: block; margin: 32px auto 0px;"
          type="button"
        >
          <span>
            Request an invite
          </span>
        </button>
      </div>
    </main>
    <footer>
      <div>
        <p>
          Made whit ðŸ–¤ in Melbourne.
        </p>
        <p>
          Â© 2016 Broccoli & Co. All rights reserved.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
